### PR TITLE
Describe when form expressions are evaluated

### LINKED ISSUE #685: Expressions in the whole form are evaluated from form open to close on every screen change
https://docs.opendatakit.org/form-logic/#when-expressions-are-evaluated is not quite right.

The mental model to have is that the **entire form** is re-evaluated every time the form is opened, a different question is shown, the form is saved, the form is closed. This is not exactly accurate -- in fact only expressions dependent on expressions whose values have changed are re-evaluated but that's an implementation detail for added efficiency. https://github.com/opendatakit/docs/pull/413 really helps detail how this is done behind the scenes and it would be useful to find a place for it for advanced users and developers.

### PR DESCRIPTION
**Pull request by lognaturel**
Closes #685

### CODE REVIEW

**Comment by adammichaelwood** [comment]
Thank you for this.

File: odk1-src/form-logic.rst [hunk 1]
```diff
@@ -139,30 +139,37 @@ you must first use a :tc:`calculate` row and then a variable.
...
 
-Expressions are evaluated at two points:
+Every expression is constantly re-evaluated as an enumerator progresses through a form. This is an important mental model to have and can explain sometimes unexpected behavior. More specifically, expressions are re-evaluated when:
 
-- when the form is advanced to the widget that contains the expression
-- when the form is saved
+- a form is opened
+- the value of any question in the form changes
+- a repeat group is added or deleted
+- a form is finalized
```
Comment (yanokwa) [inline odk1-src/form-logic.rst]: s/finalized/saved or finalized (finalization always saves)

File: odk1-src/form-logic.rst [hunk 2]
```diff
@@ -139,30 +139,37 @@ you must first use a :tc:`calculate` row and then a variable.
...
-the expression is evaluated when the form advances 
-to the widget after the calculation.
+A common misconception is that expressions are only evaluated when a question that uses it is displayed. This faulty mental model leads form designers to include functions such as :tc:`random()` or :tc:`now()` and to expect them to be evaluated exactly once. In fact, they will be re-evaluated over and over again until the form is finalized. For example, the following calculate will keep track of the last time the form was finalized:
+
+.. csv-table:: survey
+  :header: type, name, label, calculation
+
+  calculate, now, , now()
+
+The :tc:`once()` function inhibits redundant evaluation by only evaluating the expression passed into it if the node has no value. That means the expression will be evaluated once either on form load or when any values the expression depends on are set.
```
Comment (yanokwa) [inline odk1-src/form-logic.rst]: s/inhibits redundant/prevents multiple/
s/load/open/

**Comment by lognaturel** [comment]
Thanks @yanokwa! Done -- please squash commits if happy.

**Pull request merged**